pstoremem/addr_book.go

```diff
@@ -252,7 +252,7 @@ func (mab *memoryAddrBook) addAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Du
 	// }
 
 	// if we've expired all the signed addresses for a peer, remove their signed routing state record
-	if len(addrs) == 0 {
+	if len(amap) == 0 {
 		delete(s.signedPeerRecords, p)
 	}
```

Comment:
I'm not sure what the original code was trying to do anyway.

I think for the code to really do what the comment claims to do, it would have to:

1. Go through all addresses in the peer record.
2. Check if we still have them.
3. If we don't, remove the signed record.

This just clears the record if we have no more addresses. But the method that actually _adds_ addresses to a peer shouldn't be the one doing this; it should be done by the GC routine.

Reply:
@yusefnapora Do we need to clear this record in `addAddrs` for some reason ? If not, I can get rid of this code path completely.

Reply:
you guys are right; this isn't needed and was likely leftover from when we only had one type of addr. thanks for catching @aarshkshah1992